### PR TITLE
Feature/unit tests/infra 30094/standard lib

### PR DESCRIPTION
**Pull request by kkania-splunk**
ut for standard_lib modules:

- addon_parser/savedsaerches_parser.py
- event_ingestors/file_monitor_ingestor.py
- requirement_tests/test_generator.py
- app_test_generator.py

### CODE REVIEW

File: pytest_splunk_addon/standard_lib/app_test_generator.py [hunk 1]
```diff
@@ -120,11 +120,8 @@ def generate_tests(self, fixture):
                         test_type="line_breaker"
                         )
                 )
-            if isinstance(pytest_params, str):
-                LOGGER.warning(pytest_params)
 
-            elif pytest_params:
-                yield from sorted(pytest_params, key=lambda param: param.id)
+            yield from sorted(pytest_params, key=lambda param: param.id)
```
Comment (kkania-splunk) [inline pytest_splunk_addon/standard_lib/app_test_generator.py:124]: I decided to remove unreachable code, let me know if you see the need for additional exception handling

File: tests/unit/tests_standard_lib/test_event_ingestors/test_file_monitor_ingestor.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        fmei.create_event_file(
+            sample_event(
+                event="CREATE EVENT",
+                metadata={"host": "localhost", "source": "sys.log"},
+            )
+        )
+        open_mock.assert_has_calls(
+            [
+                call("fake_file", "w+"),
+                call().__enter__(),
```
Comment (uoboda-splunk) [inline tests/unit/tests_standard_lib/test_event_ingestors/test_file_monitor_ingestor.py:94]: I think checking `__enter__` and `__exit__` calls ca be removed

**Comment by uoboda-splunk** [comment]
LGTM

File: tests/unit/tests_standard_lib/test_event_ingestors/test_file_monitor_ingestor.py (hunk 2 above)
Comment (kkania-splunk) [reply tests/unit/tests_standard_lib/test_event_ingestors/test_file_monitor_ingestor.py:94]: as discussed I will leave it as it is

**Pull request merged**